### PR TITLE
fix DictReader definition

### PR DESCRIPTION
**Pull request by graingert**
* constructable with Iterable
* supports `__next__` on python 3
* supports `next` on python 2

### CODE REVIEW

**Comment by graingert** [comment]
and I've updated the documentation on cpython accordingly: https://github.com/python/cpython/pull/2653

File: stdlib/2and3/csv.pyi [hunk 1]
```diff
@@ -72,11 +72,14 @@ else:
...
+        def __init__(self, f: Iterable[str], fieldnames: Sequence[str] = ...,
                      restkey: Optional[str] = ..., restval: Optional[str] = ..., dialect: _Dialect = ...,
                      *args: Any, **kwds: Any) -> None: ...
-        def __iter__(self) -> Iterator[OrderedDict[Any, str]]: ...
-        def next(self) -> OrderedDict[Any, str]: ...
+        def __iter__(self) -> 'DictReader': ...
+        if sys.version_info >= (3,):
+            def __next__(self) -> OrderedDict[Any, str]: ...
+        else:
+            def next(self) -> OrderedDict[Any, str]: ...
```
Comment (graingert) [inline stdlib/2and3/csv.pyi]: Wondering if these should be regular dicts on python < 3.6

**Comment by gvanrossum** [comment]
Does this need to be included in 0.521? If so we need to do a typeshed sync
in mypy.

**Comment by JelleZijlstra** [comment]
I think so, there was a regression since 0.511 (DictReader could not be instantiated).

**Comment by gvanrossum** [comment]
Oh, that's serious indeed. IIRC I ran into that too and tried to fix it (86070643ac783c87f1e763b9859da8c2b23c95c1) but didn't test it for Python 3. :-( I guess the commit that caused all this (1c8498228da2ffb5b6fb735c4237b36d99679fd1) was a bit careless about it.

**Pull request merged**